### ISSUE TITLE
`generic.ci.security.use-frozen-lockfile.use-frozen-lockfile-yarn` causes false positives

### ISSUE THREAD

**Issue by ianmuscat** [returntocorp/sgrep-rules#2560]
**Describe the bug**
`generic.ci.security.use-frozen-lockfile.use-frozen-lockfile-yarn` causes false positives when `--frozen-lockfile` is not the first argument after `yarn install`.

**To Reproduce**
```
RUN yarn install --production --frozen-lockfile
```
```
RUN yarn install --prod --frozen-lockfile --prefer-offline --ignore-optional --no-progress
```

**Expected behavior**
The rule should not trigger since `--frozen-lockfile` is being specified

**Priority**
How important is this to you?
- **P2**: annoying but not blocking me

**Comment by ianmuscat** [comment]
On a related note, `--frozen-lockfile` seems to be deprecated in favour of  `--immutable`, meaning that the `generic.ci.security.use-frozen-lockfile.use-frozen-lockfile-yarn` check also results in false positives when `--immutable` is used.

>For backward compatibility we offer an alias under the name of `--frozen-lockfile`, but it will be removed in a later release.

See https://yarnpkg.com/cli/install#details for more information.

**Comment by 0xDC0DE** [comment]
Thanks for reporting this issue! I have updated the rule to use the dockerfile parser instead of the generic parser. This should allow me to be a bit more specific regarding the use of flags. The issues should be fixed when this gets released.

I've also changed the message and fix to reflect the recommendation of `--immutable` instead of `--frozen-lockfile`. Thanks for the suggestion, @ianmuscat !
